File: frontend/src/api/school.ts
import axios from './axios'
import type { Result, PageResult } from './axios'

export interface School {
  schoolId: string
  schoolName: string
  schoolShortName: string
  province: string
  schoolType: string
  contactName: string
  contactPhone: string
  contactEmail: string
  website?: string
  remark?: string
  status: string
  createdAt?: string
}

export interface SchoolQuery {
  keyword?: string
  province?: string
  schoolType?: string
  status?: string
  page?: number
  pageSize?: number
}

export interface CreateSchoolRequest {
  schoolName: string
  schoolShortName: string
  province: string
  schoolType: string
  contactName: string
  contactPhone: string
  contactEmail: string
  website?: string
  remark?: string
}

export type UpdateSchoolRequest = CreateSchoolRequest

export const getSchools = (params?: SchoolQuery) =>
  axios.get<Result<PageResult<School>>>('/v1/admin/schools', { params })

export const createSchool = (data: CreateSchoolRequest) =>
  axios.post<Result<{ schoolId: string }>>('/v1/admin/schools', data)

export const updateSchool = (schoolId: string, data: UpdateSchoolRequest) =>
  axios.put(`/v1/admin/schools/${schoolId}`, data)

export const toggleSchoolStatus = (schoolId: string, status: string) =>
  axios.patch(`/v1/admin/schools/${schoolId}/status`, { status })

File: backend/src/main/java/com/campus/platform/entity/School.java
package com.campus.platform.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("school")
public class School extends BaseEntity {

    @TableId(type = IdType.INPUT, value = "school_id")
    private UUID schoolId;

    @TableField("school_name")
    private String schoolName;

    @TableField("school_short_name")
    private String schoolShortName;

    private String province;

    @TableField("school_type")
    private String schoolType;

    @TableField("contact_name")
    private String contactName;

    @TableField("contact_phone")
    private String contactPhone;

    @TableField("contact_email")
    private String contactEmail;

    private String website;

    private String remark;

    private String status;

    @TableField("created_by")
    private UUID createdBy;
}
